proxy_agent/src/proxy/authorization_rules.rs

```diff
@@ -37,107 +29,198 @@ pub struct AuthorizationRules {
-                            privileges,
-                            identities,
-                        });
+                        Some(privilege_assignments)
                     }
-                    Some(rules)
-                }
-                None => None,
-            },
-            None => None,
+                    None => None,
+                };
+                (identities, privileges, privilege_assignments)
+            }
+            None => (None, None, None),
         };
 
         AuthorizationRules {
             defaultAllowed: authorization_item.defaultAccess.to_lowercase() == "allow",
             mode: authorization_item.mode.to_lowercase(),
-            rules,
+            identities,
+            privileges,
+            privilegeAssignments: privilege_assignments,
+        }
+    }
+
+    fn vec_to_dict<T>(vec: Option<Vec<T>>) -> Option<HashMap<String, T>>
+    where
+        T: Clone + Keyable,
+    {
+        match vec {
+            Some(vec) => {
+                let mut dict: HashMap<String, T> = HashMap::new();
+                for v in vec {
+                    dict.insert(v.get_key(), v.clone());
+                }
+                Some(dict)
+            }
+            None => None,
+        }
+    }
+
+    fn contains_key<T>(dict: &Option<HashMap<String, T>>, key: String) -> bool {
+        match dict {
+            Some(dict) => dict.contains_key(&key),
+            None => false,
+        }
+    }
+
+    fn get_from_dict<T>(dict: &Option<HashMap<String, T>>, key: String) -> Option<T>
+    where
+        T: Clone,
+    {
+        match dict {
+            Some(dict) => dict.get(&key).cloned(),
+            None => None,
         }
     }
 
     pub fn is_allowed(&self, connection_id: u128, request_url: hyper::Uri, claims: Claims) -> bool {
         if self.mode.to_lowercase() == "disabled" {
+            Connection::write_information(
+                connection_id,
+                "Access control is in disabled state, skip....".to_string(),
+            );
+
             return true;
         }
 
-        if let Some(rules) = &self.rules {
-            let mut role_privilege_matched = false;
-            for rule in rules {
-                // is privilege match
-                for privilege in &rule.privileges {
-                    if privilege.is_match(connection_id, request_url.clone()) {
-                        role_privilege_matched = true;
-                        for identity in &rule.identities {
-                            if identity.is_match(connection_id, claims.clone()) {
-                                return true;
+        if let (Some(privileges), Some(identities), Some(privilege_assignments)) = (
+            &self.privileges,
+            &self.identities,
+            &self.privilegeAssignments,
+        ) {
+            let mut privilege_matched = false;
+            for privilege in privileges.values() {
+                let privilege_name = privilege.name.clone();
```

Comment:
use borrowing as the rust error suggested

```
cannot move out of `privilege.name` which is behind a shared reference
move occurs because `privilege.name` has type `std::string::String`, which does not implement the `Copy` traitrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B1%5D?1#file%3A%2F%2F%2Fc%253A%2Fsrc%2Fgithub.com%2Fzhidongpeng%2FGuestProxyAgent%2Fproxy_agent%2Fsrc%2Fproxy%2Fauthorization_rules.rs)
authorization_rules.rs(170, 34): consider borrowing here: `&`
authorization_rules.rs(170, 48): consider cloning the value if the performance cost is acceptable: `.clone()`
```